File: WdtcCore/src/main/java/org/wdt/wdtc/core/download/source/BmclDownloadSource.java
package org.wdt.wdtc.core.download.source;

import org.wdt.wdtc.core.download.infterface.DownloadSourceInterface;

public class BmclDownloadSource implements DownloadSourceInterface {
  @Override
  public String getLibraryUrl() {
    return "https://bmclapi2.bangbang93.com/maven/";
  }

  @Override
  public String getAssetsUrl() {
    return "https://bmclapi2.bangbang93.com/assets/";
  }

  @Override
  public String getMetaUrl() {
    return getOfficialUrl();
  }

  @Override
  public String getDataUrl() {
    return getOfficialUrl();
  }

  @Override
  public String getVersionManifestUrl() {
    return "https://bmclapi2.bangbang93.com/mc/game/version_manifest.json";
  }

  @Override
  public String getFabricMetaUrl() {
    return "https://bmclapi2.bangbang93.com/fabric-meta/";
  }

  @Override
  public String getFabricLibraryUrl() {
    return "https://bmclapi2.bangbang93.com/maven/";
  }

  @Override
  public String getForgeLibraryMavenUrl() {
    return "https://bmclapi2.bangbang93.com/maven/";
  }

  @Override
  public String getVersionClientUrl() {
    return "https://bmclapi2.bangbang93.com/version/%s/%s";
  }

  @Override
  public String getOfficialUrl() {
    return "https://bmclapi2.bangbang93.com/";
  }
}

File: WdtcCore/src/main/kotlin/org/wdt/wdtc/core/download/source/BmclDownloadSource.kt
package org.wdt.wdtc.core.download.source

import org.wdt.wdtc.core.download.infterface.DownloadSourceInterface

class BmclDownloadSource : DownloadSourceInterface {
  override val libraryUrl: String
    get() = "https://bmclapi2.bangbang93.com/maven/"
  override val assetsUrl: String
    get() = "https://bmclapi2.bangbang93.com/assets/"
  override val metaUrl: String
    get() = officialUrl
  override val dataUrl: String
    get() = officialUrl
  override val versionManifestUrl: String
    get() = "https://bmclapi2.bangbang93.com/mc/game/version_manifest.json"
  override val fabricMetaUrl: String
    get() = "https://bmclapi2.bangbang93.com/fabric-meta/"
  override val fabricLibraryUrl: String
    get() = libraryUrl
  override val forgeLibraryMavenUrl: String
    get() = libraryUrl
  override val versionClientUrl: String
    get() = "https://bmclapi2.bangbang93.com/version/%s/%s"
  override val officialUrl: String
    get() = "https://bmclapi2.bangbang93.com/"
  // TODO: Bmcl quilt api is not available - see https://bmclapidoc.bangbang93.com/
  override val quiltMetaUrl: String
    get() = "https://bmclapi2.bangbang93.com/quilt-meta/"
  override val quiltMavenUrl: String
    get() = libraryUrl

}
